go/caddy/pfpki/pfpki.go

```diff
@@ -0,0 +1,144 @@
+package pfpki
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/inverse-inc/packetfence/go/caddy/caddy"
+	"github.com/inverse-inc/packetfence/go/caddy/caddy/caddyhttp/httpserver"
+	"github.com/inverse-inc/packetfence/go/db"
+	"github.com/inverse-inc/packetfence/go/log"
+	"github.com/inverse-inc/packetfence/go/panichandler"
+	"github.com/inverse-inc/packetfence/go/sharedutils"
+	"github.com/jinzhu/gorm"
+	"golang.org/x/text/message"
+	"golang.org/x/text/message/catalog"
+)
+
+// Register the plugin in caddy
+func init() {
+	caddy.RegisterPlugin("pfpki", caddy.Plugin{
+		ServerType: "http",
+		Action:     setup,
+	})
+	dict, err := parseYAMLDict()
+	if err != nil {
+		panic(err)
+	}
+	cat, err := catalog.NewFromMap(dict)
+	if err != nil {
+		panic(err)
+	}
+	message.DefaultCatalog = cat
+}
+
+type (
+	// Handler struct
+	Handler struct {
+		Next   httpserver.Handler
+		router *mux.Router
+		DB     *gorm.DB
+		Ctx    context.Context
+	}
+)
+
+// Setup the pfpki middleware
+func setup(c *caddy.Controller) error {
+	ctx := log.LoggerNewContext(context.Background())
+
+	pfpki, err := buildPfpkiHandler(ctx)
+
+	sharedutils.CheckError(err)
+
+	httpserver.GetConfig(c).AddMiddleware(func(next httpserver.Handler) httpserver.Handler {
+		pfpki.Next = next
+		return pfpki
+	})
+
+	return nil
+}
+
+func buildPfpkiHandler(ctx context.Context) (Handler, error) {
+
+	pfpki := Handler{}
+
+	Database, err := gorm.Open("mysql", db.ReturnURI(ctx, "pf"))
+	sharedutils.CheckError(err)
+
+	gorm.DefaultTableNameHandler = func(Database *gorm.DB, defaultTableName string) string {
+		return "pki_" + defaultTableName
+	}
+	if log.LoggerGetLevel(ctx) == "DEBUG" {
+		pfpki.DB = Database.Debug()
+	} else {
+		pfpki.DB = Database
+	}
+
+	pfpki.Ctx = ctx
+
+	// Default http timeout
+	http.DefaultClient.Timeout = 10 * time.Second
+
+	pfpki.router = mux.NewRouter()
+	PFPki := &pfpki
+	api := pfpki.router.PathPrefix("/api/v1").Subrouter()
+
+	// CAs (GET: list, POST: create)
+	api.Handle("/pki/cas", getSetCA(PFPki)).Methods("GET", "POST")
```

Comment:
Is it possible you're missing a file in the PR ?

I don't see getSetCA defined in any file here.

Same for a few other functions I tested below

Reply:
https://github.com/inverse-inc/packetfence/blob/feature/pfpki/go/caddy/pfpki/handlers.go#L83

Reply:
#semaanisblind